desc/equilibrium/equilibrium.py

```diff
@@ -353,15 +374,34 @@ def __init__(
 
         # ensure number of field periods agree before setting guesses
         eq_NFP = self.NFP
+        eq_NFP_umbilic_factor = self.NFP_umbilic_factor
         surf_NFP = self.surface.NFP if hasattr(self.surface, "NFP") else self.NFP
+        surf_NFP_umbilic_factor = (
+            self.surface.NFP_umbilic_factor
+            if hasattr(self.surface, "NFP_umbilic_factor")
+            else self.NFP_umbilic_factor
+        )
         axis_NFP = self._axis.NFP
+        axis_NFP_umbilic_factor = self._axis.NFP_umbilic_factor
         errorif(
             not (eq_NFP == surf_NFP == axis_NFP),
             ValueError,
             "Unequal number of field periods for equilibrium "
             + f"{eq_NFP}, surface {surf_NFP}, and axis {axis_NFP}",
         )
 
+        errorif(
+            not (
+                eq_NFP_umbilic_factor
+                == surf_NFP_umbilic_factor
+                == axis_NFP_umbilic_factor
+            ),
+            ValueError,
+            "Unequal number of umbilic field period factors for equilibrium "
```

Comment:
this should just be "unequal umbilici field period prefactor" or something, not number of